### ISSUE TITLE
query DSL

### ISSUE THREAD

**Issue by MarkNicholls** [Zaid-Ajaj/Snowflaqe#25]
Hello,

not a problem, just questions.
I've run this against a hot chocolate server, and worked a treat...but the generated project is hard wired into Fable? What if I just want a vanilla core application?

I'm also interested to know, is there a DSL or maybe there's a DSL available in another library (hmmm..not sure how that would work), but I don't want to write Graphql and then generate/validate it, I want to write queries in F# that are consistent with the schema, and then have them generate graphql invisibly.

**Comment by MarkNicholls** [comment]
ah just noticed the documentation about the non fable implementation I'll give that a go.

still interested in a DSL though...i.e. more like EF than dapper.

**Comment by Zaid-Ajaj** [comment]
Hi there, glad to hear Snowflaqe worked for you 😄 

About the DSL, I don't think I have any plans to implement one. I understand why but I believe the GraphQL syntax is a good DSL on its own where we don't need another layer of complexity in the project. I also like the fact that you can just copy and paste the query you have from any GraphQL playground and have Snowflaqe generate the code accordingly. 

In conclusion, a F# DSL won't add much value to the current implementation (sorry!)

**Comment by MarkNicholls** [comment]
Ok, fair enough.
So you're not interested if I wrote one? which I probably wont, but you never know.

**Comment by Zaid-Ajaj** [comment]
I don't think a DSL would work in the first place because the query has to be available at build time but a F# DSL needs to be interpreted at runtime to generate the underlying GraphQL query first before actually generating the corresponding code from Snowflaqe. Unless you don't want to actually have typed responses but that goes against the idea of this project

**Comment by MarkNicholls** [comment]
I was envisaging snowflaqe just generates the model, like scaffolding a EF model from a database.
The value (to me) is that when someone changes the server, I just rescaffold the model, all the queries I've defined that no longer "work" don't compile....I think graphql queries are functors, where the maps are costrained to be maps to child attributes or queries on references, you could do it in IQueryable, except it would be easy to write queries that can't be expressed in graphql....but you'd probably learn the hard way you have no "bind", or you do "bind" client side.
don't know....just freewheeling at the moment.

**Comment by Zaid-Ajaj** [comment]
It is technically possible to do this: take the server schema and generate a DSL from which queries can written as `IQueryable` but that is so much more work than we already have here with the same benefits: run Snowflaqe again to re-generate the client to see whether the queries are correct and whether anything has breaking changes. The project won't compile if fields are removed or their types changed. It is an interesting idea nonetheless 😄
